### PR TITLE
Feature: add jest configuration and scripts with testing for dab_client and dab_adb_bridge

### PR DESCRIPTION
**Pull request by andrewqiamzn**
# Why?
The DAB-ADB bridge currently has no tests. We want to add unit tests with Jest and achieve minimum 80% coverage. This pull request covers:
- [x] Jest set up and configuration
- [x] Unit testing for `dab_client.ts` (88% line coverage)
- [x] Unit testing for `dab_adb_bridge.ts` (95.53% line coverage)

# Acceptance Criteria
- [x] 80% coverage with unit tests

# Test Plan
- [x] E2E testing
lSMguQXqWl 06 11

- [x] Unit testing
JY8aH6N6Br 11 57

### CODE REVIEW

File: adapters/dab-adb-bridge/src/lib/util.ts [hunk 1]
```diff
@@ -33,38 +30,3 @@ export function adbAppStatusToDabAppState(appStatus: AndroidApplicationStatus) {
             return ApplicationState.Stopped;
     }
 }
-
-// Static logger
```
Comment (andrewqiamzn) [inline adapters/dab-adb-bridge/src/lib/util.ts:37]: This is so that we can mock it easier in the future

File: adapters/dab-adb-bridge/src/config/test.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (andrewqiamzn) [inline adapters/dab-adb-bridge/src/config/test.json:1]: Would get `WARNING: NODE_ENV value of 'test' did not match any deployment config file names` when testing while importing a config file 
ShpUcSf05Y 03 09

Comment (andrewqiamzn) [reply adapters/dab-adb-bridge/src/config/test.json:1]: Adding an empty test.json file was the fix
https://github.com/node-config/node-config/wiki/Strict-Mode

File: adapters/dab-adb-bridge/src/lib/__tests__/dab_adb_bridge.test.ts [hunk 3]
```diff
@@ -0,0 +1,792 @@
...
+            };
+
+            const actualRestartDeviceResponse = await testAdbBridge.restartDevice();
+
+            expect(mockMqttClient.publish).toBeCalledWith(topics.DAB_MESSAGES, {
+                timestamp: +new Date(),
+                level: "warn",
+                message: "Device is rebooting and will be temporarily offline"
+            });
+            // expect(mockAdbCommands.reboot).toBeCalled();
```
Comment (andrewqiamzn) [inline adapters/dab-adb-bridge/src/lib/__tests__/dab_adb_bridge.test.ts:585]: Struggled to test this reboot part. Not sure why it is never called in my test suite. Can anyone help me with this?

File: adapters/dab-adb-bridge/src/lib/dab/__tests__/dab_client.test.ts [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    });
+
+    describe("exitApp", () => {
+        it.each`
+            forceInputAtDabClient | expectedForceInputAtMqtt
+            ${true}               | ${true}
+            ${false}              | ${false}
+            ${undefined}          | ${false}
+        `(
+            "should return the state as $expectedGetAppState if the app is $appStatus",
```
Comment (giotar) [inline adapters/dab-adb-bridge/src/lib/dab/__tests__/dab_client.test.ts:161]: The variable names are wrong

File: adapters/dab-adb-bridge/src/lib/__tests__/dab_adb_bridge.test.ts [hunk 5]
```diff
@@ -0,0 +1,792 @@
...
+
+        it("should start adb without any parameters", async () => {
+            const launchAppRequest = {
+                appId: "Settings"
+            };
+            const expectedLaunchAppResponse = {
+                status: 200
+            };
+            const mockDabResponseCall = jest.fn(() => expectedLaunchAppResponse);
+            testAdbBridge.dabResponse = mockDabResponseCall;
```
[the remark this answers is not in the document]
Comment (andrewqiamzn) [reply adapters/dab-adb-bridge/src/lib/__tests__/dab_adb_bridge.test.ts:304]: Oh good spot! Will remove in my next commit.

File: adapters/dab-adb-bridge/src/lib/__tests__/dab_adb_bridge.test.ts [hunk 6]
```diff
@@ -0,0 +1,792 @@
...
+
+            const actualLaunchAppResponse = await testAdbBridge.launchApp(launchAppRequest);
+
+            expect(mockAdbCommands.start).toBeCalledWith(["intent1", "intent2"]);
+            expect(mockDabResponseCall).toBeCalled();
+            expect(actualLaunchAppResponse).toEqual(expectedLaunchAppResponse);
+        });
+
+        it("should start adb with the right list of intents (with a parameter appended at the end) when one parameter is added", async () => {
```
[the remark this answers is not in the document]
Comment (andrewqiamzn) [reply adapters/dab-adb-bridge/src/lib/__tests__/dab_adb_bridge.test.ts:313]: Yes, there's a flow just for where there's multiple parameters, so this test covers the single parameter flow ⬇️ 
Screenshot 2022-12-08 at 10 03 01

File: adapters/dab-adb-bridge/src/lib/dab/__tests__/dab_client.test.ts (hunk 4 above)
Comment (andrewqiamzn) [reply adapters/dab-adb-bridge/src/lib/dab/__tests__/dab_client.test.ts:161]: should I change `$appStatus` to `$AndroidApplicationStatus` and `$expectedGetAppState` to `$ApplicationState` in order to match the variable names to the enums?

Comment (giotar) [reply adapters/dab-adb-bridge/src/lib/dab/__tests__/dab_client.test.ts:161]: It's the whole text that is wrong, it should be something like "should call MQTT exit with force: $expectedForceInputAtMqtt if exitApp is called with force:$forceInputAtDabClient"

Comment (andrewqiamzn) [reply adapters/dab-adb-bridge/src/lib/dab/__tests__/dab_client.test.ts:161]: Oh i see what you mean. Thanks, will fix in the next commit!

**Review by giotar — APPROVED** [review]
